install/generator/04-syndesis-oauth-proxy.yml.mustache

```diff
@@ -75,9 +75,7 @@
           image: ${SYNDESIS_REGISTRY}/{{Images.OAuthProxyImagePrefix}}/{{ Images.Support.OAuthProxy }}:{{ Tags.OAuthProxy }}{{/WithDockerImages}}
           args:
             - --provider=openshift
-{{^WithOAuthClient}}
-            - --client-id=system:serviceaccount:${OPENSHIFT_PROJECT}:syndesis-oauth-client{{/WithOAuthClient}}{{#WithOAuthClient}}
-            - --client-id=syndesis-${OPENSHIFT_PROJECT}{{/WithOAuthClient}}
+            - --client-id=system:serviceaccount:${OPENSHIFT_PROJECT}:syndesis-oauth-client
             - --client-secret=${OPENSHIFT_OAUTH_CLIENT_SECRET}
```

Comment:
iiur this env var OPENSHIFT_OAUTH_CLIENT_SECRET is not needed when using an OAuthClient and could have been removed from the templates parameters ? (this is just a question to validate my understanding of how it worked)

Reply:
I just checked, so currently, this variable is currently generated automatically and reused when creating the OAuthClient. Howeve, I think it should be mandatory and the template should fail if this secret is not set (so removing the auto generation). Because the required service account will for sure have a different token. So the advantage would be that the template cannot be instantiated if the token is not provided, and this more or less requires the user to create the SA first and extract the token. Otherwise, the installation will just fail with an error that the SA is missing (as it has to be created from the outside). We could add some minor instructions to this template parameter so its clear that an SA has to be created in advance.

But maybe one could add a token alread to an SA, too ? Then we could the SA directly here like you did with the OAuthClient.

Just thinking a loud. Said that, I'm happy as it is as the templates are going away eventually anyway.